MetaMask/metamask-extension
# Disable fiat conversion in Approve Screen when `Show balance and token price checker` is OFF

## Explanation

Fixes #17428 
Fiat value was getting displayed on the approve screen when `Show balance and token price checker` was disabled, so we are adding the `useCurrencyRateCheck` to the Approve screen to control the Fiat display.


## Screenshots/Screencaps



### Before



### After
<img width="358" alt="Screenshot 2023-01-26 at 4 11 23 PM" src="https://user-images.githubusercontent.com/43930900/214951301-9b4538fd-696a-4dee-adc1-fa519001fd84.png">



## Manual Testing Steps
Disable the Fiat conversion on Settings.

For ERC721:

1. Go to https://bscscan.com/token/0xADc466855ebe8d1402C5F7e6706Fccc3AEdB44a0#writeContract
2. Connect MM
3. Cliick Write Contract
4. Click Approve function
5. Add address and id (any number)
6. Click Write
7. See MM prompt and fees displayed in fiat

For ERC20:

1. Go to https://bscscan.com/token/0x2170ed0880ac9a755fd29b2688956bd959f933f8#writeContract
2. Connect MM
3. Cliick Write Contract
4. Click Approve function
5. Add address and amount (any number)
6. Click Write
7. See MM prompt and fees displayed in fiat



## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
